src/base/serial/fossil.c

```diff
@@ -86,10 +86,17 @@ boolean fossil_tsr_installed = FALSE;
 static u_short irq_hlt;
 static void fossil_irq(Bit16u idx, void *arg);
 
-static int fossil_init(void)
+static int fossil_init(int *info)
```

Comment:
See what I mean, now you made it
multi-purpose and renamed to `info`. :)
I think it would be cleaner to do the
installed and no-ports checks prior
to calling `fossil_init()` and simply not
call it in case of an error. Then you
can even make it `void`.